packages/indexer/src/starknet/parser.ts

```diff
@@ -0,0 +1,50 @@
+import { FieldElement } from "./felt";
+
+interface EventFilterParser {
+  eventFilter(name: string): Filter;
+  // eventParser(name: string): ParsedEvent[];
+}
+
+interface Filter {
+  name: string;
+}
+
+interface Event {
+  // Parsed event properties
+  event: any;
+  transaction: any;
+}
+
+export class Contract implements EventFilterParser {
+  readonly contractAddress: FieldElement;
+  readonly contractAbi: any[];
+
+  constructor(contractAddress: FieldElement, contractAbi: any[]) {
+    this.contractAddress = contractAddress;
+    this.contractAbi = contractAbi;
+  }
+
+  eventFilter(name: string): Filter {
+    // Check if event exists in ABI
+    const event = this.contractAbi.find((item) => item.name === name);
+    if (!event) {
+      throw new Error(`Event ${name} not found in contract ABI`);
+    }
+
+    // Need more clarity on what the DNA filter should look like
```

Comment:
The `EventFilter` type I linked above is how we define filters in DNA.

 * `fromAddress`: contract address
 * `keys`: is an array of field elements (encoded as hex). At the moment, most (all?) events have a single key, which is the Pedersen hash of the event name. For example, the key for the `Transfer` event is `keys: ["0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9"]`. Using starknet.js, you can compute the hash as follows:

```ts
import { hash } from "starknet"

hash.getSelectorFromName("Transfer") // => "0x99..."
```

 * `data`: event data. Not used, leave unset.
 * `includeReverted`: should we include reverted transactions? Not implemented, so leave unset.
 * `includeTransaction`: should it include the transaction that generated the event? Since we don't need it, set this to `false`.
 * `includeReceipt`: should it include the receipt of the transaction that generated the event? Set to `false` as well.